### PR TITLE
Refactor packaging test for data-dependencies

### PR DESCRIPTION
**Pull request by hurryabit**
We have plenty of tests that check that a certain feature can be used
with data-dependencies. Most of these tests have exactly the same
structure and have most likely been copied and pasted. Let's put an end
to this and use a function abstracting this test pattern instead. I'll
add quite a few more tests of this kind, so the gains will be even
bigger than shown here.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/7664)

### CODE REVIEW

File: compiler/damlc/tests/src/DA/Test/Packaging.hs [hunk 1]
```diff
@@ -1987,6 +1757,37 @@ dataDependencyTests Tools{damlc,repl,validate,davlDar,oldProjDar} = testGroup "D
...
+            step "building project to be imported via data-dependencies"
+            createDirectoryIfMissing True (tmpDir </> "type")
+            writeFileUTF8 (tmpDir </> "type" </> "daml.yaml") $ unlines
+                [ "sdk-version: " <> sdkVersion
+                , "name: type"
+                , "source: ."
+                , "version: 0.1.0"
+                , "dependencies: [daml-prim, daml-stdlib]"
+                ]
+            writeFileUTF8 (tmpDir </> "type" </> "P1.daml") $ unlines importee
```
Comment (cocreature) [inline compiler/damlc/tests/src/DA/Test/Packaging.hs:1773]: `P1` and `P2` seems unnecessarily confusing. Why not `Dep` and `Main` or `Importer` and `Importee`?

**Review by cocreature — APPROVED** [review]
Nice cleanup, thanks!

File: compiler/damlc/tests/src/DA/Test/Packaging.hs (hunk 1 above)
Comment (hurryabit) [reply compiler/damlc/tests/src/DA/Test/Packaging.hs:1773]: I thought about that as well. I'll change it to `Lib` and `Main`, I'm worried `Importer` and `Importee` are too close to get mixed up.

Comment (hurryabit) [reply compiler/damlc/tests/src/DA/Test/Packaging.hs:1773]: The `automerge` label was faster than me... Here we go: https://github.com/digital-asset/daml/pull/7665

Comment (hurryabit) [reply compiler/damlc/tests/src/DA/Test/Packaging.hs:1773]: I'm not gaming my PR stats!

Comment (cocreature) [reply compiler/damlc/tests/src/DA/Test/Packaging.hs:1773]: looking at the contributor stats you should :trollface:

**Pull request merged**